lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -473,25 +469,39 @@ class BuildOpGroups : public calyx::FuncOpPartialLoweringPattern {
     assert(
         lhsIsSeqOp != rhsIsSeqOp &&
         "unexpected sequential operation on both sides; please open an issue");
-    Operation *seqOp = calyx::parentIsSeqCell(cmpIOp.getLhs())
-                           ? cmpIOp.getLhs().getDefiningOp()
-                           : cmpIOp.getRhs().getDefiningOp();
-    condReg = getState<ComponentLoweringState>().getSeqResReg(seqOp);
+    // If `cmpIOp`'s lhs/rhs operand is the result of a sequential operation,
+    // its result will be stored in a register.
+    resReg =
+        cast<calyx::RegisterOp>(lhsIsSeqOp ? cmpIOp.getLhs().getDefiningOp()
+                                           : cmpIOp.getRhs().getDefiningOp());
 
     auto groupOp = cast<calyx::GroupOp>(group);
     getState<ComponentLoweringState>().addBlockScheduleable(cmpIOp->getBlock(),
                                                             groupOp);
 
-    buildAssignmentsForRegisterWrite(
-        rewriter, groupOp, getState<ComponentLoweringState>().getComponentOp(),
-        resReg, condReg.getOut());
+    rewriter.setInsertionPointToEnd(groupOp.getBodyBlock());
+    auto loc = cmpIOp.getLoc();
+    // `calyxOp` will be one of `EqLibOp`, `SleLibOp`, etc. The
+    // 2'th `result` corresponds to the output result of `calyxOp`.
```

Comment:
I'd prefer this to be an `assert`ion rather than a comment, i.e., can we assert this type matches?